src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/TraceUtilities.cs

```diff
@@ -44,6 +68,10 @@ public static MonitoringSourceConfiguration GetTraceConfiguration(Models.TracePr
 
                 configurations.Add(new MetricSourceConfiguration(options.GetIntervalSeconds(), defaultProviders));
             }
+            if (profile.HasFlag(Models.TraceProfile.Gc))
+            {
+                configurations.Add(new GcCollectConfiguration());
```

Comment:
IIRC our current merging logic relies on having mutually exclusive configurations across different profiles. Please double check that combined profiles work as expected when they share a provider (cpu + gc comes to mind since I believe they both use Microsoft-Windows-DotNETRuntime)

Reply:
While the [diagnostics code](https://github.com/dotnet/diagnostics/blob/fe7763cdbd0ccbec8304887130bf3514a7f4f087/src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/AggregateSourceConfiguration.cs#L22) says about de-duplicating the providers, it isn't strictly necessary.

The [runtime code](https://github.com/dotnet/runtime/blob/34d13b27a4f0d62367145bcc1f67f7044ad2328a/src/native/eventpipe/ep-config.c#L65) will perform the proper merging anyway, it will take the bitwise or for the keyword and take the max for the level. It needs to do that when there are multiple tools connected to it anyway.

And the combination of profiles works (in the sense that we get both types of events)